Source/hydro/advection_util.cpp

```diff
@@ -237,31 +237,64 @@ Castro::divu(const Box& bx,
     Real vy = 0.0_rt;
 
     if (coord_type == 0) {
-      ux = 0.5_rt * (q_arr(i,j,k,QU) - q_arr(i-1,j,k,QU) + q_arr(i,j-1,k,QU) - q_arr(i-1,j-1,k,QU)) * dxinv;
-      vy = 0.5_rt * (q_arr(i,j,k,QV) - q_arr(i,j-1,k,QV) + q_arr(i-1,j,k,QV) - q_arr(i-1,j-1,k,QV)) * dyinv;
+
+        // Cartesian
+
+        ux = 0.5_rt * (q_arr(i,j,k,QU) - q_arr(i-1,j,k,QU) + q_arr(i,j-1,k,QU) - q_arr(i-1,j-1,k,QU)) * dxinv;
+        vy = 0.5_rt * (q_arr(i,j,k,QV) - q_arr(i,j-1,k,QV) + q_arr(i-1,j,k,QV) - q_arr(i-1,j-1,k,QV)) * dyinv;
+
+    } else if (coord_type == 1) {
+
+        // Cylindrical R-Z
+
+        if (i == 0) {
+            ux = 0.0_rt;
+            vy = 0.0_rt;  // is this part correct?
+        } else {
+            Real rl = (i - 0.5_rt) * dx[0] + problo[0];
+            Real rr = (i + 0.5_rt) * dx[0] + problo[0];
+            Real rc = (i) * dx[0] + problo[0];
+
+            // These are transverse averages in the y-direction
+            Real ul = 0.5_rt * (q_arr(i-1,j,k,QU) + q_arr(i-1,j-1,k,QU));
+            Real ur = 0.5_rt * (q_arr(i,j,k,QU) + q_arr(i,j-1,k,QU));
+
+            // Take 1/r d/dr(r*u)
+            ux = (rr * ur - rl * ul) * dxinv / rc;
+
+            // These are transverse averages in the x-direction
+            Real vb = 0.5_rt * (q_arr(i,j-1,k,QV) + q_arr(i-1,j-1,k,QV));
+            Real vt = 0.5_rt * (q_arr(i,j,k,QV) + q_arr(i-1,j,k,QV));
+
+            vy = (vt - vb) * dyinv;
+        }
 
     } else {
-      if (i == 0) {
-        ux = 0.0_rt;
-        vy = 0.0_rt;  // is this part correct?
-      } else {
+
+        // Spherical R-Theta
+
         Real rl = (i - 0.5_rt) * dx[0] + problo[0];
```

Comment:
I don't think that these are correct

I think they should be

```
rl = i * dx[0] + problo[0]
rr = (i  + 1.0) * dx[0] + problo[0]
rc = (i + 0.5) * dx[0] + problo[0]
```

that makes rc "cell-centered"

Reply:
oh... wait... this is supposed to be node centered though